pkg/vmprovider/providers/vsphere2/vmprovider_vm.go

```diff
@@ -942,23 +943,23 @@ func (vs *vSphereVMProvider) vmCreateGenConfigSpecExtraConfig(
 	vmCtx context.VirtualMachineContextA2,
 	createArgs *VMCreateArgs) error {
 
-	ecMap := make(map[string]string, len(vs.globalExtraConfig))
+	ecMap := maps.Clone(vs.globalExtraConfig)
 
-	// The only use of this template is for the JSON_EXTRA_CONFIG that is set in gce2e env
-	// to populate {{.ImageName }} so vcsim will create a container for the VM.
-	// BMV: This should be removable now that vcsim gce2e is gone.
-	renderTemplateFn := func(name, text string) string {
-		t, err := template.New(name).Parse(text)
-		if err != nil {
-			return text
-		}
-		b := strings.Builder{}
-		if err := t.Execute(&b, createArgs.ImageStatus); err != nil {
-			return text
+	if v, exists := ecMap[constants.ExtraConfigRunContainerKey]; exists {
+		// For local-vcsim the JSON_EXTRA_CONFIG is set and contains RUN.container so we use the
+		// template to fill in the image {{.Name }} so vcsim will create a container for the VM.
+		renderTemplateFn := func(name, text string) string {
+			t, err := template.New(name).Parse(text)
+			if err != nil {
+				return text
+			}
+			b := strings.Builder{}
+			if err := t.Execute(&b, createArgs.ImageStatus); err != nil {
```

Comment:
If we ever template more fields we should define an intermediate struct

Reply:
Maybe leave a TODO in the code instead of on the review?